### PR TITLE
Move exports to main source file

### PR DESCRIPTION
**Pull request by vincentmolin**
An attempt at implementing #394

### CODE REVIEW

**Comment by mcabbott** [comment]
This looks great, +1 for organising them by file.

File: src/batched/batchedmul.jl [hunk 1]
```diff
@@ -1,7 +1,3 @@
-
-export batched_mul, batched_mul!, ⊠,  batched_vec
-export batched_transpose, batched_adjoint
-
 include("./batchedadjtrans.jl")
```
Comment (mcabbott) [inline src/batched/batchedmul.jl:1]: Can I suggest also moving this `include` to the main file?

**Comment by CarloLucibello** [comment]
Can I suggest a small extension of this PR and move also the `using`s to the main file?

**Comment by vincentmolin** [comment]
@CarloLucibello moved most occurences to the top of `NNlib.jl`, left
https://github.com/FluxML/NNlib.jl/blob/aa868277a0faec74ffa7b0cebadac38c4bd0e76e/src/nnpack/NNPACK.jl#L1
and
https://github.com/FluxML/NNlib.jl/blob/aa868277a0faec74ffa7b0cebadac38c4bd0e76e/src/gemm.jl#L7
intact, as well as any in the testfiles.

**Comment by vincentmolin** [comment]
Moved most occurences of `using` to the top of `NNlib.jl`, left
https://github.com/FluxML/NNlib.jl/blob/aa868277a0faec74ffa7b0cebadac38c4bd0e76e/src/nnpack/NNPACK.jl#L1
and
https://github.com/FluxML/NNlib.jl/blob/aa868277a0faec74ffa7b0cebadac38c4bd0e76e/src/gemm.jl#L7
intact, as well as any in the testfiles.

**Pull request merged**